internal/characters/cyno/attack.go

```diff
@@ -0,0 +1,138 @@
+package cyno
+
+import (
+	"fmt"
+
+	"github.com/genshinsim/gcsim/internal/frames"
+	"github.com/genshinsim/gcsim/pkg/core/action"
+	"github.com/genshinsim/gcsim/pkg/core/attributes"
+	"github.com/genshinsim/gcsim/pkg/core/combat"
+)
+
+var (
+	attackFrames          [][]int
+	attackHitmarks        = [][]int{{14}, {17}, {13, 22}, {27}}
+	attackHitlagHaltFrame = [][]float64{{0.01}, {0.06}, {0, 0.02}, {0.04}}
+	attackDefHalt         = [][]bool{{false}, {true}, {false, true}, {true}}
+)
+
+const normalHitNum = 4
+
+func init() {
+	attackFrames = make([][]int, normalHitNum)
+	attackFrames[0] = frames.InitNormalCancelSlice(attackHitmarks[0][0], 28) // N1 -> N2
+	attackFrames[0][action.ActionAttack] = 15
+
+	attackFrames[1] = frames.InitNormalCancelSlice(attackHitmarks[1][0], 23) // N2 -> N3
+	attackFrames[1][action.ActionAttack] = 22
+
+	attackFrames[2] = frames.InitNormalCancelSlice(attackHitmarks[2][1], 27) // N3 -> N4
+	attackFrames[2][action.ActionCharge] = 26
+
+	attackFrames[3] = frames.InitNormalCancelSlice(attackHitmarks[3][0], 58) // N4 -> N1
+	attackFrames[3][action.ActionCharge] = 500                               // impossible action
+}
+
+func (c *char) Attack(p map[string]int) action.ActionInfo {
+	if c.StatusIsActive(burstKey) {
```

Comment:
This part feels very hacky. I think your best option would be to:
- have both c.NormalCounter and c.NormalBCounter 
- overwrite AdvanceNormalAttackIndex, ResetNormalCounter and NextNormalCounter and then just operate on one of the counters based on whether burst is active (ResetNormalCounter needs to reset both though).